### DISCUSSION TITLE
Are we ready for next release?

### DISCUSSION THREAD

**Discussion by hazendaz** [spotbugs/spotbugs#2880] [General]
Its been a few months, lots of false positive fixes, I think we are probably ready for a release unless others want to get some more items in place.  Thoughts?

**Comment by hazendaz** [comment]
Lets do this by next week.  Moving dates around.  I want to do so bit of cleanup as I'm seeing on other repos where I'm using a sane usage of eclipse to cleanup code that many that seemed good have many issues and figure its worth the extra effort to cleanup a bit.  Also think we need to look at some items stuck in draft that could go with little effort.

**Comment by JuditKnoll** [comment]
The only thing that concerns me is the reintroduction of https://github.com/spotbugs/spotbugs/issues/2686. Otherwise, I think we are good to go with 4.8.4.

**Reply by ThrawnCA** [reply]
Is there no way to pin the versions we use?

**Reply by hazendaz** [reply]
well we intentionally updated because we thought it was fixed so we need to roll back parts of it :)

**Reply by gtoison** [reply]
I've made a PR to revert the commons-text upgrade: https://github.com/spotbugs/spotbugs/pull/2892

**Reply by hazendaz** [reply]
ok so that one is fixed.

**Comment by hazendaz** [comment]
So now, we have maybe 3 PRs that could come in that are out there.  They all have questions back to the users.  Think if they are not critical and I'm not so sure they are, we can just cut release then those can go next time.

I know @JuditKnoll had concern with that old PR with stale bot that I enabled.  I get it as far as concerns of losing things but seems it won't go stale if assigned and the stale marker that is on most of them isn't on some of the oldest.  Maybe because I had rebased some branches.  Not sure there.  I presume the stale marker survives to closed.  But keep in mind we have some that should be marked stale still from 2017.  Its kind of clear to me those are not getting worked.  And while we might miss some that get closed, they are easily searched so if important to someone they would search.  I think our PRs state to do so.  Typically I don't like those type of bots if too aggressive but think its not too aggressive.  However, we can make it less aggressive.  Or we could just remove it.  I'll leave that to everyone else to decide.  To me it just seemed helpful to cut down more on open items as I feel that gives a perception to users that maybe we don't really do much.

Anyway, I'd like to get a release this weekend if at all possible.

**Reply by gtoison** [reply]
I'm going a bit off-topic here but I'm not a big fan of the automation for stale issues and PR. For the most part these stale issues are valid but were never investigated due to the lack of time.

**Reply by JuditKnoll** [reply]
I agree with @gtoison about the stale issue and PR automation. 
Also, thank you for the fix.
I see no issue blocking the release.

**Reply by hazendaz** [reply]
Feel free to remove the stale GHA.  I don't like it generally either so seems we all agree we don't want it.

**Reply by hazendaz** [reply]
Stale items are actually easy to find since they are tagged.  Just go to search and search on 'stale', it will show them all.  Still, if we want to remove that go for it.  I think maybe keep half of it.  Let it flag stale but not remove.  I can reopen all the ones it closed otherwise.  Not letting that hold up release though as we were just sitting on the release.  I went ahead and released as I needed java 23 support that comes with the release.

**Reply by hazendaz** [reply]
All stale PRs that were closed have been reopened and the latest update will just mark them stale but will not close any going forwards.  While they were easy to find by just looking for 'label:stale', it will be better now as we discussed to just keep them alive.

**Comment by hazendaz** [comment]
Released 4.8.4

**Reply by iloveeclipse** [reply]
Open issue regarding release: https://github.com/spotbugs/spotbugs/issues/2921
